x/datapool/keeper/pool_test.go

```diff
@@ -183,6 +191,309 @@ func (suite *poolTestSuite) TestGetPool() {
+
+	shareToken, err := suite.DataPoolKeeper.SellData(suite.Ctx, requesterAddr, *cert)
+	suite.Require().Error(err)
+	suite.Require().Equal(types.ErrPoolNotFound, err)
+	suite.Require().Nil(shareToken)
+}
+
+func (suite *poolTestSuite) TestSellData_impossible_status_pool() {
+	poolID := uint64(1)
+	round := uint64(1)
+	dataHash := []byte("dataHash")
+
+	validatorAccount := suite.AccountKeeper.NewAccountWithAddress(suite.Ctx, dataVal1)
+	err := validatorAccount.SetPubKey(dataValPubKey)
+	suite.Require().NoError(err)
+	suite.AccountKeeper.SetAccount(suite.Ctx, validatorAccount)
+
+	// Already activate status
+	pool := makeTestDataPool(poolID)
+	pool.Status = types.ACTIVE
+	suite.DataPoolKeeper.SetPool(suite.Ctx, pool)
+
+	cert, err := makeTestDataCertificate(suite.Cdc.Marshaler, poolID, round, dataHash)
+	suite.Require().NoError(err)
+
+	shareToken, err := suite.DataPoolKeeper.SellData(suite.Ctx, requesterAddr, *cert)
+	suite.Require().Error(err)
+	fmt.Println(err)
+	suite.Require().True(strings.Contains(err.Error(), "the status of the pool is not 'PENDING'"))
+	suite.Require().True(strings.HasSuffix(err.Error(), types.ErrInvalidDataValidationCert.Error()))
+	suite.Require().Nil(shareToken)
+}
+
+func (suite *poolTestSuite) TestSellData_mismatch_certificate_and_pool_round() {
+	poolID := uint64(1)
+	// Wrong rounds recorded in the certificate
+	round := uint64(2)
+	dataHash := []byte("dataHash")
+
+	validatorAccount := suite.AccountKeeper.NewAccountWithAddress(suite.Ctx, dataVal1)
+	err := validatorAccount.SetPubKey(dataValPubKey)
+	suite.Require().NoError(err)
+	suite.AccountKeeper.SetAccount(suite.Ctx, validatorAccount)
+
+	pool := makeTestDataPool(poolID)
+	suite.DataPoolKeeper.SetPool(suite.Ctx, pool)
+
+	cert, err := makeTestDataCertificate(suite.Cdc.Marshaler, poolID, round, dataHash)
+	suite.Require().NoError(err)
+
+	shareToken, err := suite.DataPoolKeeper.SellData(suite.Ctx, requesterAddr, *cert)
+	suite.Require().Error(err)
+	suite.Require().True(strings.Contains(err.Error(), "pool round do not matched. pool round: 1"))
+	suite.Require().True(strings.HasSuffix(err.Error(), types.ErrInvalidDataValidationCert.Error()))
+	suite.Require().Nil(shareToken)
+}
+
+func makeTestDataCertificate(marshaler codec.Marshaler, poolID, round uint64, dataHash []byte) (*types.DataValidationCertificate, error) {
+	unsignedCert := types.UnsignedDataValidationCertificate{
+		PoolId:        poolID,
+		Round:         round,
+		DataHash:      dataHash,
+		DataValidator: dataVal1.String(),
+		Requester:     requesterAddr.String(),
+	}
+
+	bz, err := marshaler.MarshalBinaryBare(&unsignedCert)
+	if err != nil {
+		return nil, err
+	}
+
+	sign, err := dataValPrivKey.Sign(bz)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.DataValidationCertificate{
+		UnsignedCert: &unsignedCert,
+		Signature:    sign,
+	}, nil
+}
+
+func makeTestDataPool(poolID uint64) *types.Pool {
+	nftPrice := sdk.NewCoin(assets.MicroMedDenom, sdk.NewInt(1000000))
```

Comment:
You can use `NFTPrice` declared in const.

Reply:
Okay. I will use this.